org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -6057,20 +6059,49 @@ public boolean validForeignKeys() {
 					int pocid = DB.getSQLValue(get_TrxName(), sql.toString(), fkval);
 					if (pocid < 0) {
 						log.saveError("Error", "Foreign ID " + fkval + " not found in " + fkcol);
-						return false;
+						throw new AdempiereException("Foreign ID " + fkval + " not found in " + fkcol);
```

Comment:
I think this is inconsistent with existing practise where we don't use saveError and throw exception together.

Reply:
Hi @hengsin,

Diego asked me about removing that and I suggested to keep it because the people that uses the boolean validForeignKeys can have access to the error using peekError or retrieveError, and it could be that there is already code using that approach.

Reply:
Got it @hengsin, I just added a commit to address that